unit_tests/test_redis_aggregator.py

```diff
@@ -0,0 +1,238 @@
+
+    @staticmethod
+    def test_constructor_connect_and_subscribe_called():
+        with patch("gsy_e_sdk.redis_aggregator."
+                   "RedisAggregator._connect_and_subscribe") as mocked_method:
+            RedisAggregator(aggregator_name=TEST_AGGREGATOR_NAME)
+            mocked_method.assert_called()
+
+    @staticmethod
+    def test_constructor_lock_instantiated(aggregator):
+        assert aggregator.lock is not None
+
+
+@pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+class TestRedisAggregatorDeleteAggregator:
+    """Test cases for RedisAggregator's delete_aggregator method."""
+
+    @staticmethod
+    def test_delete_aggregator_redis_db_publishes_data(aggregator):
+        data = {"name": TEST_AGGREGATOR_NAME,
+                "aggregator_uuid": TEST_AGGREGATOR_UUID,
+                "type": "DELETE",
+                "transaction_id": TEST_TRANSACTION_ID}
+
+        aggregator.aggregator_uuid = TEST_AGGREGATOR_UUID
+
+        aggregator.delete_aggregator()
+        aggregator.redis_db.publish.assert_called_with(
+            "aggregator", json.dumps(data))
+
+    @staticmethod
+    def test_delete_aggregator_api_time_out_raise_exception(aggregator):
+        with patch("gsy_e_sdk.redis_aggregator.wait_until_timeout_blocking",
+                   side_effect=AssertionError):
+            aggregator.aggregator_uuid = TEST_AGGREGATOR_UUID
+            with pytest.raises(RedisAggregatorAPIException):
+                aggregator.delete_aggregator()
+
+    @staticmethod
+    @pytest.mark.parametrize("test_is_blocking, expected_ret_val",
+                             [(True, TEST_TRANSACTION_ID), (False, None)])
+    def test_delete_aggregator_returns_expected(aggregator, test_is_blocking,
+                                                expected_ret_val):
+        aggregator.aggregator_uuid = TEST_AGGREGATOR_UUID
+        assert aggregator.delete_aggregator(is_blocking=test_is_blocking) is expected_ret_val
+
+
+@pytest.mark.usefixtures("mock_client_command_buffer_attributes")
+class TestRedisAggregatorExecuteBatchCommands:
+    """Test cases for RedisAggregator's execute_batch_commands method."""
+
+    @staticmethod
+    def test_execute_batch_commands_commands_buffer_length_zero_return_none(aggregator):
+        with patch("gsy_e_sdk.redis_aggregator.ClientCommandBuffer.buffer_length",
+                   new_callable=PropertyMock, return_value=0):
+            assert aggregator.execute_batch_commands() is None
+
+    @staticmethod
+    def test_execute_batch_commands_not_all_uuids_in_selected_device_uuid_list_raise_exception(
+            aggregator):
+        aggregator.device_uuid_list = [TEST_DEVICE_UUID_1]
+
+        with pytest.raises(Exception):
+            aggregator.execute_batch_commands(is_blocking=False)
+
+    @staticmethod
+    def test_execute_batch_commands_batched_command_published(aggregator):
+        uuid.uuid4 = MagicMock(return_value=TEST_TRANSACTION_ID)
```

Comment:
Can we reuse the `fixture_mock_transaction_id_and_timeout_blocking` here?

Reply:
Indeed

Reply:
Yes it is possible. However, at this stage of the _sub-tests_, the `timeout_blocking` mock is not yet needed and therefore would be unnecessarily mocked here. 
It is also not perjudicial, as it would not affect nothing just having it there.
I can use the fixture here so it looks cleaner, but I'll leave that call to you.